wurst/_handles/Group.wurst

```diff
@@ -65,20 +64,23 @@ public function group.getRandomUnit() returns unit
 	end)
 	return randomSelected
 	
+/* Group iterator */
+
+unit iterUnit
+group iterGroup
 
-// group iterator
 public function group.iterator() returns group
-	bj_groupAddGroupDest = CreateGroup()
-	ForGroup(this, function GroupAddGroupEnum)
-	return bj_groupAddGroupDest
+	iterGroup = getGroup()
+	ForGroup(this, () -> iterGroup.addUnit(GetEnumUnit()))
+	return iterGroup
 
 public function group.hasNext() returns boolean
-	return FirstOfGroup(this) != null
+	iterUnit = FirstOfGroup(this)
```

Comment:
this will break if .next() isn't called instantly after hasNext or if someone only calls .next()
```
while itr1.hasNext()
    if itr2.hastNext()
        let u1 = itr1.next()
        let u2 = itr2.next()  // u1 and u2 will be equal

OR


someGroup.next() // ignore one unit
let second = someGroup.next() // this one will still be the first one

```
I know these use cases shouldn't happe and the user should call .next() only, but instantly after .hasNext() anyways, but yeah :/